src/v2/codegen/emitter.rs

```diff
@@ -211,6 +211,18 @@ pub trait Emitter: Sized {
         let state = self.state();
         state.reset_internal_fields();
 
+        let m = state.get_meta();
+        if let Some(mut meta) = m.borrow_mut().clone() {
```

Comment:
This looks like we're setting title and version only if crate meta is already set. We should do it the other way. If crate meta isn't set, then we should create one from the API, since we know that the API always has title and version?

Reply:
Thanks @wafflespeanut for the feedback :) I have a few question on ☝️ 
1) Do you mean we need to check `state.crate_meta.borrow().is_none` by making `crate_meta` public ?
2) Will this check be using `cli` feature flag ?  🤔

Reply:
We have `get_meta` and `set_meta` methods for getting/setting. If `cli` isn't enabled, then whatever we do will be no-op, so you shouldn't have to worry about `cli` flag.